drivers/adc/adc_ad4114.c

```diff
@@ -0,0 +1,471 @@
+
+	/* Configure gain to 0x400000 */
+	gain[0] = 0x40;
+	gain[1] = 0x00;
+	gain[2] = 0x00;
+	ad4114_write_reg(dev, AD4114_GAIN0_REG, gain, 3);
+	ad4114_write_reg(dev, AD4114_GAIN1_REG, gain, 3);
+
+	/* Bit 6: DATA_STAT = 1 */
+	write_reg[0] = 0x0;
+	write_reg[1] = 0x40;
+	ad4114_write_reg(dev, AD4114_IFMODE_REG, write_reg, 2);
+
+	/* Bit 12: BI_UNIPOLARx = 0
+	 * Bit 9:8 INBUFx = 11
+	 */
+	write_reg[0] = 0x3;
+	write_reg[1] = 0x0;
+	ad4114_write_reg(dev, AD4114_SETUPCON0_REG, write_reg, 2);
+
+	/* Bit 12: BI_UNIPOLARx = 1
+	 * Bit 9:8 INBUFx = 11
+	 */
+	write_reg[0] = 0x13;
+	write_reg[1] = 0x0;
+	ad4114_write_reg(dev, AD4114_SETUPCON1_REG, write_reg, 2);
+
+	/* Bit 15: REF_EN = 1
+	 * Bit 3:2: CLOCKSEL = 11
+	 */
+	write_reg[0] = 0x80;
+	write_reg[1] = 0xC;
+	ad4114_write_reg(dev, AD4114_MODE_REG, write_reg, 2);
+
+	tid = k_thread_create(&data->thread, data->stack, K_KERNEL_STACK_SIZEOF(data->stack),
+			      (k_thread_entry_t)adc_ad4114_acquisition_thread, data, NULL, NULL,
+			      CONFIG_ADC_AD4114_ACQUISITION_THREAD_PRIO, 0, K_NO_WAIT);
+
+	if (IS_ENABLED(CONFIG_THREAD_NAME)) {
+		err = k_thread_name_set(tid, "adc_ad4114");
+		if (err < 0) {
+			return err;
+		}
+	}
+
+	adc_context_unlock_unconditionally(&data->ctx);
+	return 0;
+}
+
+static DEVICE_API(adc, adc_ad4114_api) = {
+	.channel_setup = adc_ad4114_channel_setup,
+	.read = adc_ad4114_read,
+};
+
+#define ADC_AD4114_FILL_MAP_INPUT(inst)                                                            \
+	{                                                                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 0),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 1),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 2),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 3),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 4),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 5),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 6),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 7),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 8),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 9),                                          \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 10),                                         \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 11),                                         \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 12),                                         \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 13),                                         \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 14),                                         \
+		DT_INST_PROP_BY_IDX(inst, map_inputs, 15),                                         \
+	}
```

Comment:
This could be simplified to `#define ADC_AD4114_FILL_MAP_INPUT DT_INST_PROP(inst, map_inputs)` and may be checked with a `BUILD_ASSERT(DT_INST_PROP_LEN(inst, map_inputs) == 16)` to be safe :)

Reply:
Thanks for the suggestion, I have changed it